File: app/src/test/java/org/schulcloud/mobile/data/model/HomeworkTest.java
package org.schulcloud.mobile.data.model;

import org.junit.Before;
import org.junit.Test;

import static junit.framework.Assert.assertEquals;

public class HomeworkTest {
    private static final String ID = "ID";
        private static final String SCHOOLDID = "schoolId";
        private static final String TEACHERID = "teacherId";
        private static final String NAME = "name";
        private static final String DESCRIPTION = "description";
        private static final String AVAILABLEDATE = "availabledate";
        private static final String DUEDATE = "dueDate";
        private static final Course COURSEID = new Course();
        private static final Boolean RESTRICTED = true;

        private Homework homework;

        @Before
        public void setUp() throws Exception {
            homework = createNewHomework();
        }

        @Test
        public void testGetProperties() {
            assertEquals(homework._id, ID);
            assertEquals(homework.schoolId, SCHOOLDID);
            assertEquals(homework.teacherId, TEACHERID);
            assertEquals(homework.name, NAME);
            assertEquals(homework.description, DESCRIPTION);
            assertEquals(homework.availableDate, AVAILABLEDATE);
            assertEquals(homework.dueDate, DUEDATE);
            assertEquals(homework.courseId, COURSEID);
            assertEquals(homework.restricted, RESTRICTED);
        }

    public static Homework createNewHomework() {
        Homework h = new Homework();
        h._id = ID;
        h.schoolId = SCHOOLDID;
        h.teacherId = TEACHERID;
        h.name = NAME;
        h.description = DESCRIPTION;
        h.availableDate = AVAILABLEDATE;
        h.dueDate = DUEDATE;
        h.courseId = COURSEID;
        h.restricted = RESTRICTED;

        return h;
    }
}

File: app/src/test/java/org/schulcloud/mobile/models/homework/HomeworkTest.kt
/*
package org.schulcloud.mobile.models.homework

import io.realm.RealmList
import org.junit.Assert.*
import org.junit.Before
import org.junit.Test
import org.schulcloud.mobile.models.base.RealmString

class HomeworkTest {
    private companion object {
        const val ID = "id"
        const val TEACHERID = "teacherId"
        const val TITLE = "title"
        const val DESCRIPTION = "description"
        const val DUEDATE = "dueDate"
        val COURSE = HomeworkCourse()
        const val RESTRICTED = true
        const val PUBLICSUBMISSIONS = true
        const val INVALID_DUEDATE = "invDueDate"
        const val VALID_DUEDATE = "2020-07-12T10:10:10.001Z"
        const val USERID_TEACHER = "teacherId"
        const val USERID_TEACHER_OTHER = "teacherIdOther"
        val COURSE_SUBSTITUTION_TEACHER = HomeworkCourse().apply {
            substitutionIds = RealmList(RealmString("teacherIdOther"))
        }
        */
/*val DUEDATE_DAY_DIFFERENCE = DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
                .print(LocalDateTime.now().plusDays(2))
        val DUEDATE_HOUR_DIFFERENCE = DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
                .print(LocalDateTime.now().plusHours(3).plusMinutes(1))*//*

        const val DUETIMESPANDAYS = 2
        const val DUETIMESPANHOURS = 3
    }

    private val newHomework: Homework
        get() = Homework().apply {
            id = ID
            teacherId = TEACHERID
            title = TITLE
            description = DESCRIPTION
            dueDate = DUEDATE
            course = COURSE
            restricted = RESTRICTED
            publicSubmissions = PUBLICSUBMISSIONS
        }
    private lateinit var homework: Homework

    @Before
    fun setUp() {
        homework = newHomework
    }

    @Test
    fun testGetProperties() {
        assertEquals(ID, homework.id)
        assertEquals(TEACHERID, homework.teacherId)
        assertEquals(TITLE, homework.title)
        assertEquals(DESCRIPTION, homework.description)
        assertEquals(DUEDATE, homework.dueDate)
        assertEquals(COURSE, homework.course)
        assertEquals(RESTRICTED, homework.restricted)
        assertEquals(PUBLICSUBMISSIONS, homework.publicSubmissions)
    }

    @Test
    fun testDatePropertiesNullWhenDuedateInvalid() {
        homework.dueDate = INVALID_DUEDATE
        assertNull(homework.dueDateTime)
        assertNull(homework.dueTimespanDays)
        assertNull(homework.dueTimespanHours)
    }

    @Test
    fun testDatetimeNotNullWhenDuedateValid() {
        homework.dueDate = VALID_DUEDATE
        assertNotNull(homework.dueDateTime)
    }

   */
/* @Test
    fun testDuetimespanCorrect() {
        homework.dueDate = DUEDATE_DAY_DIFFERENCE
        assertEquals(DUETIMESPANDAYS, homework.dueTimespanDays)
        homework.dueDate = DUEDATE_HOUR_DIFFERENCE
        assertEquals(DUETIMESPANHOURS, homework.dueTimespanHours)
    }*//*


    @Test
    fun testIsteacherFalseWhenUserAndCourseSubstitutionsNotTeacherid(){
        assertFalse(homework.isTeacher(""))
    }

    @Test
    fun testIsteacherTrueWhenUserOrCourseSubstitutionTeacherid(){
        assertTrue(homework.isTeacher(USERID_TEACHER))
        homework.course = COURSE_SUBSTITUTION_TEACHER
        // TODO: check if that is really the required behaviour
        assertTrue(homework.isTeacher(USERID_TEACHER_OTHER))
    }

    // TODO: add tests for canSeeSubmissions
}
*/
